packages/valory/skills/transaction_settlement_abci/rounds.py

```diff
@@ -379,17 +362,24 @@ class ValidateTransactionRound(VotingRound):
     def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Enum]]:
         """Process the end of the block."""
         # if reached participant threshold, set the result
+
         if self.positive_vote_threshold_reached:
+            # We obtain the latest tx hash from the `tx_hashes_history`.
+            # We keep the hashes sorted by their finalization time.
+            # If this property is accessed before the finalization succeeds,
+            # then it is incorrectly used.
```

Comment:
Indeed, so the event INCORRECT_SERIALIZATION is very misleading. The problem is not necessarily a serialisation issue. Taking into account discussion here https://github.com/valory-xyz/open-autonomy/pull/983/files#r901875239 I suggest calling it `NOTHING_TO_VALIDATE`.